WTG.Analyzers.Test/TestData/CodeContractsAnalyzer/Requires/Unknown/Result.cs

```diff
@@ -0,0 +1,44 @@
+using System.Diagnostics.Contracts;
+
+public class Bob : IBob
+{
+	public void Method1(int value)
+	{
+		if (value <= 3)
+		{
+			throw new System.ArgumentException("Invalid Argument.", nameof(value));
+		}
+
+		value.GetType();
+	}
+
+	protected void Method2(int value)
+	{
+		if (value <= 3)
+		{
+			throw new System.ArgumentException("Message", nameof(value));
+		}
+
+		value.GetType();
+	}
+
+	void IBob.Method(int value)
+	{
+		if (value <= 3)
```

Comment:
ohhhhhhhhhhh so _that's_ why you improved the boolean inversion.

Reply:
Yeah, but more the other cases: `if (!(value != null))` and `if (!!string.IsNullOrEmpty(value))`.